### PR TITLE
feat: Add a Kafka event type for the Lambda

### PR DESCRIPTION
**Pull request by EdwOK**
*Description of changes:*
Add a Kafka event type for the AWS Lambda.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by EdwOK** [comment]
@ganeshnj I opened PR against `dev`.

File: Libraries/src/Amazon.Lambda.KafkaEvents/Amazon.Lambda.KafkaEvents.csproj
Comment (ganeshnj) [inline Libraries/src/Amazon.Lambda.KafkaEvents/Amazon.Lambda.KafkaEvents.csproj:9]: There is no reason to start with version 2. Let's go with `1.0.0`

File: Libraries/src/Amazon.Lambda.KafkaEvents/KafkaEvent.cs
Comment (ganeshnj) [inline Libraries/src/Amazon.Lambda.KafkaEvents/KafkaEvent.cs:45]: partition is defined as string property as per https://docs.aws.amazon.com/lambda/latest/dg/with-msk.html

de-serialization failed for me when using the lambda test tool. It should be string.

That said, could you also add sample request for local debugging?

There are some example requests [here](https://github.com/aws/aws-lambda-dotnet/tree/087590ce99274e16e26d37e1dfd73b0b71d1230a/Tools/LambdaTestTool/src/Amazon.Lambda.TestTool/Resources/SampleRequests)

File: Libraries/src/Amazon.Lambda.KafkaEvents/README.md
Comment (ganeshnj) [inline Libraries/src/Amazon.Lambda.KafkaEvents/README.md:14]: I didn't fully understand purpose of this. Where is `MemoryStream` object used? Did you mean Value inside the record?

File: Libraries/test/EventsTests.Shared/kafka-event.json
Comment (ganeshnj) [inline Libraries/test/EventsTests.Shared/kafka-event.json:8]: I'm not sure whether this has to be string or int. https://docs.aws.amazon.com/lambda/latest/dg/with-kafka.html shows it as string but https://aws.amazon.com/blogs/compute/using-self-hosted-apache-kafka-as-an-event-source-for-aws-lambda/ show it as int.

I have reached out to the team to make sure we are modeling the event object right.

**Review by ganeshnj — CHANGES_REQUESTED** [review]
Again, thanks for the PR.

I did first pass and like I mentioned in one of the comment, I reached out the Kafka team internally to make sure we have right shape for the event.

File: Libraries/src/Amazon.Lambda.KafkaEvents/Amazon.Lambda.KafkaEvents.csproj
Comment (EdwOK) [reply Libraries/src/Amazon.Lambda.KafkaEvents/Amazon.Lambda.KafkaEvents.csproj:9]: Fixed.

File: Libraries/src/Amazon.Lambda.KafkaEvents/KafkaEvent.cs
Comment (EdwOK) [reply Libraries/src/Amazon.Lambda.KafkaEvents/KafkaEvent.cs:45]: Done. I added kafka event request to samples.

File: Libraries/test/EventsTests.Shared/kafka-event.json
Comment (EdwOK) [reply Libraries/test/EventsTests.Shared/kafka-event.json]: Your assumption that the `Partition` should be **integer** is [true](https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/Partition.cs#L54). 
I fixed JSON.

File: Libraries/src/Amazon.Lambda.KafkaEvents/README.md
Comment (EdwOK) [reply Libraries/src/Amazon.Lambda.KafkaEvents/README.md:14]: I think this section should be removed because there is nothing special for serialization here.

Comment (EdwOK) [reply Libraries/src/Amazon.Lambda.KafkaEvents/README.md]: Fixed.

**Comment by EdwOK** [comment]
Hello @normj, Could you take a look my PR?

**Comment by EdwOK** [comment]
@normj @boblodgett @ganeshnj

**Comment by EdwOK** [comment]
Hello @ganeshnj, How much longer do I have to wait for you to review?

**Comment by ganeshnj** [comment]
Apologies, couldn't focus on this. I will get back to this by end of day today.

File: Libraries/src/Amazon.Lambda.KafkaEvents/KafkaEvent.cs
Comment (ganeshnj) [inline Libraries/src/Amazon.Lambda.KafkaEvents/KafkaEvent.cs:46]:
```suggestion
            public string Partition { get; set; }
```

We should change the test event json and keep the Partition as string.

All the docs that I could see, they all have Partition as string https://github.com/awsdocs/aws-lambda-developer-guide/blob/afca60967a1e1a56c6ee4b6d00131b9c0dbc3f9d/doc_source/with-msk.md

Changing the expectation will to runtime exception here.

File: Libraries/test/EventsTests.Shared/kafka-event.json
Comment (ganeshnj) [inline Libraries/test/EventsTests.Shared/kafka-event.json:9]:
```suggestion
        "partition": "0",
```

File: Tools/LambdaTestTool/src/Amazon.Lambda.TestTool/Resources/SampleRequests/kafka-event.json
Comment (ganeshnj) [inline Tools/LambdaTestTool/src/Amazon.Lambda.TestTool/Resources/SampleRequests/kafka-event.json:9]:
```suggestion
        "partition": "0",
```

**Review by ganeshnj — CHANGES_REQUESTED** [review]
I think we are pretty close to merge this PR, I have one small comment.

**Review by ganeshnj — APPROVED** [review]

**Review by normj — APPROVED** [review]

**Comment by EdwOK** [comment]
Hello @boblodgett Can you please take a look?

**Comment by ganeshnj** [comment]
@EdwOK we don't really need another approval. I'm working on logistics to release this new nuget package as this requires release pipeline update. Work is scheduled for next week.

**Comment by ganeshnj** [comment]
https://www.nuget.org/packages/Amazon.Lambda.KafkaEvents/ is available.

Thanks again @EdwOK for the PR.

**Pull request merged**
